### PR TITLE
feat(machines): display error message for no machines

### PR DESCRIPTION
**Pull request by petermakowski**
## Done

- display error message for no machines 
- update selected machines on delete notify
- extract NodeActionWarning component

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Select all machines
- Open Delete action form via "take action" menu
- Make sure the delete count on the button is correct
- Open machine details for one of the machines in the list in a new tab
- Delete the machine
- Go back to the previous tab
- Make sure the delete count has been reduced by 1 to reflect the new value

---
- create 2 new machines
- add these machine hostnames as a filter to the search field (e.g. `hostname:(my-machine,other-machine)`)
- select all machines
- open delete action form via "take action" 
- Open machine details for machines you just created in a new tab
- Delete the machine
- Go back to the previous tab
- Make sure the selected count has been reduced by 1 (from `Delete 2 machines` to `Delete machine`).
- Select all machines by clicking the checkbox at the top of machine listing table
- Open the remaining machine details page in a new tab
- Delete that machine
- Make sure the "No machines selected" warning is displayed and the submit button is disabled
- Click cancel to close the form

## Fixes

Fixes: https://github.com/canonical/app-tribe/issues/1288

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

image

### CODE REVIEW

**Comment by ndv99** [comment]
In part 1 I'm guessing I need to delete a machine from its details page, that step was missed out - Selecting all machines on polong shows 852 machines (correct), but opening a new tab with machine details doesn't lower the count in the original tab on the delete button or form. However it works fine if I delete the machine in the other tab, which I'm guessing is what you meant for me to do here :joy:

**Comment by petermakowski** [comment]
@ndv99 Correct, updated the steps.

**Review by ndv99 — APPROVED** [review]
Nice one!

File: src/app/base/components/FormikFormContent/FormikFormContent.test.tsx [hunk 1]
```diff
@@ -62,6 +62,30 @@ describe("FormikFormContent", () => {
...
+                submitDisabled
+                submitLabel="Save"
+              >
+                Content
+              </FormikFormContent>
+            </Formik>
+          </CompatRouter>
+        </MemoryRouter>
+      </Provider>
+    );
```
[the remark this answers is not in the document]
Comment (ndv99) [reply src/app/base/components/FormikFormContent/FormikFormContent.test.tsx:84]: If you run with this change just remember to do the import for the render util

**Pull request merged**